### DISCUSSION TITLE
Generate components in commonMain and platform modules e.g. androidMain

### DISCUSSION THREAD

**Discussion by witmitchell** [evant/kotlin-inject#268] [Q&A] [answered]
Hi!

I was wondering if its possible to use components in commonMain and in platform specific modules like androidMain/iosMain? The reason ask this because im using kmm and Compose multiplatform so i moved all my view related code to commonMain. I want to be able to inject the context in androidMain so i need a platform specific component and i have non-platform specific components in commonMain for api calls etc. When i add components in commonMain and androidMain and hit build i get a redeclaration error because common component is also build in androidMain.
Is there a way to achieve this? Maybe scope some components to the common module only?

**Comment by evant** [comment, marked as answer]
Check out issues on the [ksp project repo](https://github.com/google/ksp/issues), particularly https://github.com/google/ksp/issues/965
